src/profile/components/Settings/Notifications/index.tsx

```diff
@@ -1,200 +1,130 @@
-    }
-  };
-
-  public subscribe = async () => {
-    const { subscription, message } = await registerPushManager();
-    this.setState({ message, subscription });
-
-    const { user } = this.context;
-    if (user && subscription) {
-      await subscribe(subscription, user);
-    }
+  const dispatchSubscribe = () => {
+    dispatch(registerDeviceForPushNotifications());
   };
 
-  public unsubscribe = async () => {
-    const { user } = this.context;
-    const sub = await getNotificationSubscription();
-    if (user && sub) {
-      await unsubscribe(sub, user);
-      const removed = await removeNotificationSubscription();
-      if (removed) {
-        this.setState({ subscription: undefined });
-      }
-    }
+  const disaptchUnsubscribe = () => {
+    dispatch(unregisterDeviceFromPushNotfications());
   };
 
-  public render() {
-    const { channels, allowNotifications, subscription, message } = this.state;
-    const { selectedChannels } = this.state;
-    return (
-      <>
-        <Pane>{ABOUT_NOTIFICATIONS}</Pane>
-        <Pane>
-          {ABOUT_BROWSER_SUPPORT}
-          <div className={style.container}>
-            <BrowserSupport name="Notification" value={verifyNotification()} />
-            <BrowserSupport name="PushManager" value={verifyPushManager()} />
-            <BrowserSupport name="ServiceWorker" value={verifyServiceWorker()} />
-          </div>
-        </Pane>
-        <Pane>
-          {ABOUT_ENABLE_NOTIFICATIONS}
-          <div className={style.container}>
-            {message && md`**${message}**`}
-            <Option
-              description="Tillat notifikasjoner på dette nettstedet"
-              name="allowNotifications"
-              value={allowNotifications}
-              toggle={this.toggleGlobalNotifications}
-            />
-            <Option
-              description="Registrer denne enheten for å motta notifikasjoner"
-              name="subscription"
-              value={!!subscription}
-              toggle={!!subscription ? this.unsubscribe : this.subscribe}
-            />
-          </div>
-        </Pane>
-        <Pane>
-          {ABOUT_NOTIFICATION_OPTIONS}
-          <div className={style.container}>
-            {channels.map((channel) => (
-              <Option
-                key={channel.name}
-                name={channel.name}
-                description={channel.description}
-                value={selectedChannels.includes(channel.name)}
-                toggle={() => this.toggleChannel(channel.name)}
-              />
-            ))}
-          </div>
-        </Pane>
-      </>
-    );
-  }
-}
+  const dispatchResolveCurrentSubscription = useCallback(() => {
+    dispatch(resolveCurrentDeviceSubscription());
+  }, [dispatch]);
+
+  const dispatcFetchPermissions = useCallback(() => {
+    dispatch(fetchPermissions());
+  }, [dispatch]);
+
+  const dispatcFetchUserPermissions = useCallback(() => {
+    dispatch(fetchUserPermissions());
+  }, [dispatch]);
```

Comment:
It is to keep referential equality of the functions which are in the dependency list of the `ùseEffect` hooks.

The functions which are not wrapped in a `useCallback` do not need to be referentially equal, since they are not used in any hooks or other memoized components.